### ISSUE TITLE
Cannot connect with username/password: BadInternalError

### ISSUE THREAD

**Issue by JianYang93** [open62541/open62541#4708]
Dear all,

I am connecting to an OPCUA server. The server side enables both anonymous and username/password login. The server enables all None, Basic128 and Basic256 security policies.

When I try to connect with anonymous login, the server can be connected. The nodes on the server side can be browsed. However due to some internal settings I can only browse but not able to read node values. The log is as follows:

Log start!!!!!!!!!!!!!
CLaserDrillingDlg::OnInitDialog, m_hWnd = 0x306b8.
DevicesManager::Init in.
DevicesManager::Init out.
[2021-10-18 19:48:56.024 (UTC+0800)] warn/userland      AcceptAll Certificate Verification. Any remote certificate will be accepted.
[2021-10-18 19:48:56.044 (UTC+0800)] info/channel       Connection 1364 | SecureChannel 4219026839 | SecureChannel opened with SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None and a revised lifetime of 600.00s
[2021-10-18 19:48:56.044 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Closed, ConnectStatus: Good
[2021-10-18 19:48:56.048 (UTC+0800)] info/client        Client Status: ChannelState: Closed, SessionState: Closed, ConnectStatus: Good
7 endpoints found
URL of endpoint 0 is opc.tcp://192.168.51.200:4840
URL of endpoint 1 is opc.tcp://192.168.51.200:4840
URL of endpoint 2 is opc.tcp://192.168.51.200:4840
URL of endpoint 3 is opc.tcp://192.168.51.200:4840
URL of endpoint 4 is opc.tcp://192.168.51.200:4840
URL of endpoint 5 is opc.tcp://192.168.51.200:4840
URL of endpoint 6 is opc.tcp://192.168.51.200:4840
[2021-10-18 19:48:57.335 (UTC+0800)] info/channel       Connection 1400 | SecureChannel 4219026840 | SecureChannel opened with SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None and a revised lifetime of 600.00s
[2021-10-18 19:48:57.336 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Closed, ConnectStatus: Good
[2021-10-18 19:48:57.440 (UTC+0800)] info/client        Rejecting UserTokenPolicy 0 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256' not available
[2021-10-18 19:48:57.440 (UTC+0800)] info/client        Rejecting UserTokenPolicy 1 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256' not available
[2021-10-18 19:48:57.440 (UTC+0800)] info/client        Rejecting UserTokenPolicy 2 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256' not available
[2021-10-18 19:48:57.441 (UTC+0800)] info/client        Rejecting UserTokenPolicy 3 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256' not available
[2021-10-18 19:48:57.441 (UTC+0800)] info/client        Selected endpoint 0 in URL opc.tcp://192.168.51.200:4840 with SecurityMode None and SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None
[2021-10-18 19:48:57.441 (UTC+0800)] info/client        Selected UserTokenPolicy Anonymous_Token with UserTokenType Anonymous and SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None
[2021-10-18 19:48:57.450 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Created, ConnectStatus: Good
[2021-10-18 19:48:57.474 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Activated, ConnectStatus: Good
[2021-10-18 19:48:57.475 (UTC+0800)] info/client        OpcUaClient connect ok!

When I try to connect with username/password, it fails. (The same set of username/password works very well on UAExpert)
The log is as follows:

Log start!!!!!!!!!!!!!
CLaserDrillingDlg::OnInitDialog, m_hWnd = 0x306a4.
DevicesManager::Init in.
DevicesManager::Init out.
[2021-10-18 19:50:04.574 (UTC+0800)] warn/userland      AcceptAll Certificate Verification. Any remote certificate will be accepted.
[2021-10-18 19:50:04.625 (UTC+0800)] info/channel       Connection 1368 | SecureChannel 4219026994 | SecureChannel opened with SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None and a revised lifetime of 600.00s
[2021-10-18 19:50:04.625 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Closed, ConnectStatus: Good
[2021-10-18 19:50:04.632 (UTC+0800)] info/client        Client Status: ChannelState: Closed, SessionState: Closed, ConnectStatus: Good
7 endpoints found
URL of endpoint 0 is opc.tcp://192.168.51.200:4840
URL of endpoint 1 is opc.tcp://192.168.51.200:4840
URL of endpoint 2 is opc.tcp://192.168.51.200:4840
URL of endpoint 3 is opc.tcp://192.168.51.200:4840
URL of endpoint 4 is opc.tcp://192.168.51.200:4840
URL of endpoint 5 is opc.tcp://192.168.51.200:4840
URL of endpoint 6 is opc.tcp://192.168.51.200:4840
[2021-10-18 19:50:08.217 (UTC+0800)] info/channel       Connection 1388 | SecureChannel 4219027004 | SecureChannel opened with SecurityPolicy http://opcfoundation.org/UA/SecurityPolicy#None and a revised lifetime of 600.00s
[2021-10-18 19:50:08.218 (UTC+0800)] info/client        Client Status: ChannelState: Open, SessionState: Closed, ConnectStatus: Good
[2021-10-18 19:50:08.224 (UTC+0800)] info/client        Rejecting UserTokenPolicy 0 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256' not available
[2021-10-18 19:50:08.224 (UTC+0800)] info/client        Rejecting UserTokenPolicy 1 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256' not available
[2021-10-18 19:50:08.224 (UTC+0800)] info/client        Rejecting UserTokenPolicy 2 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256' not available
[2021-10-18 19:50:08.224 (UTC+0800)] info/client        Rejecting UserTokenPolicy 3 in endpoint 0: security policy 'http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256' not available
[2021-10-18 19:50:08.225 (UTC+0800)] info/client        Rejecting UserTokenPolicy 4 (anonymous) in endpoint 0: configuration doesn't match
[2021-10-18 19:50:08.226 (UTC+0800)] info/client        Rejecting endpoint 1: security policy not available
[2021-10-18 19:50:08.227 (UTC+0800)] info/client        Rejecting endpoint 2: security policy not available
[2021-10-18 19:50:08.229 (UTC+0800)] info/client        Rejecting endpoint 3: security policy not available
[2021-10-18 19:50:08.229 (UTC+0800)] info/client        Rejecting endpoint 4: security policy not available
[2021-10-18 19:50:08.229 (UTC+0800)] info/client        Rejecting endpoint 5: security policy not available
[2021-10-18 19:50:08.229 (UTC+0800)] info/client        Rejecting endpoint 6: security policy not available
[2021-10-18 19:50:08.230 (UTC+0800)] error/client       No suitable UserTokenPolicy found for the possible endpoints
[2021-10-18 19:50:08.230 (UTC+0800)] info/client        Client Status: ChannelState: Closed, SessionState: Closed, ConnectStatus: BadInternalError

Basically I did not set any configurations but only used the default config. The functions I used were UA_Client_connect and UA_Client_connectUsername respectively.

I really appreciate any help! Thanks in advance!
